src/mixins/java/org/spongepowered/common/mixin/core/world/entity/player/PlayerMixin.java

```diff
@@ -616,4 +618,18 @@ public void attack(final Entity targetEntity) {
     public boolean impl$canCallIgniteEntityEvent() {
         return !this.shadow$isSpectator() && !this.shadow$isCreative();
     }
+
+    @Inject(method = "canHarmPlayer", at = @At("HEAD"), cancellable = true)
+    private void impl$onCanHarmPlayer(final net.minecraft.world.entity.player.Player other, final CallbackInfoReturnable<Boolean> cir) {
+        if (!(other instanceof org.spongepowered.api.entity.living.player.server.ServerPlayer)) {
+            return;
+        }
+
+        //Check whatever the other player can hit this player
+        //Fixes per player scoreboards which could have different team set up for each player
+        final Team otherTeam = other.getTeam();
+        final Team thisTeam = ((net.minecraft.world.scores.Scoreboard) ((ServerPlayerBridge) other).bridge$getScoreboard()).getPlayersTeam(this.shadow$getScoreboardName());
```

Comment:
Still not convinced by this.

I wanted to setup a test case for this the other day but it seems adding/removing a player from a team is broken.

Reply:
I made a sample plugin to test this: https://gist.github.com/aromaa/3c43abec63a8a800e056246c532e3cf5

Connect two clients to the server and run /scoreboardteambug. When this patch is applied, the player who sees the prefix [FF TRUE] can hit the other player but not the player with [FF FALSE]. With the current api-8 branch BOTH players can hit each other, which is incorrect.